### ISSUE TITLE
Visual problems after updating to Ubuntu 21.10

### ISSUE THREAD

**Issue by JezerM** [awesomeWM/awesome#3469]
Output of `awesome --version`:

```
awesome v4.3-1220-g1d55ae09 (Too long)
 • Compiled against Lua 5.3.3 (running with Lua 5.3)
 • API level: 4
 • D-Bus support: yes
 • xcb-errors support: no
 • execinfo support: yes
 • xcb-randr version: 1.6
 • LGI version: 0.9.2
 • Transparency enabled: yes
 • Custom search paths: no
```

**How to reproduce the issue:**

I recently updated Ubuntu from 21.04 to 21.10, and tried to compile AwesomeWM for the first time.

**Actual result:**

Wiboxes doesn't seem to work as expected:
- Buttons dissapears
- Inverted shapes
- Icons with weird sizes

This is a screenshot of how it looks:
![Screenshot_2021-10-16-00_1366x768](https://user-images.githubusercontent.com/59768785/137604997-e6de320b-e7d4-4a81-9330-3cbc97441712.png)

And here a video demonstrating the result: https://streamable.com/87vdw8

**Expected result:**

- Powerline shapes should point the other way around
- Left buttons on wibar shouldn't dissapear

Screenshot before updating (months ago):
![Screenshot_2021-07-27-42_1366x768](https://user-images.githubusercontent.com/59768785/137605062-83e51125-a02f-4956-9188-d36718b06918.png)

I guess this may be related to dependency conflicts. Also, tried to recompile with Lua3.1, Lua3.3, LuaJIT, and remove/install with **apt**, but the problem persists.

**Comment by Elv13** [comment]
Hello,

Looks like you have a compositing manager like picom running. Can you confirm it works fine when you disable it? If so, the problem is there.

Otherwise, the version ubuntu ships hasn't changed in a few years, so this isn't due to a new feature

**Comment by JezerM** [comment]
Yep, I'm using Picom, but disabling it just makes transparent things to be black. Nothing related.

**Comment by JezerM** [comment]
Although, AwesomeWM was working fine after updating. The problem came when I compiled it, but no errors or warnings. So, idk where the problem could be.

**Comment by Elv13** [comment]
Can you bisect which commit causes the regression? (that means recompiling some random commits between the point where it works and the point where it doesn't. `git bisect` and Xephyr helps with that).

**Comment by JezerM** [comment]
Tried bisect and compile: 70cd2938 was the first one that didn't crash, but the problem is still there. Newer commits work the same... I'm starting to think the problem is Ubuntu.

![Screenshot_2021-10-16-15_640x480](https://user-images.githubusercontent.com/59768785/137612399-1da35ec9-f621-4924-a336-d14469691643.png)

**Comment by JezerM** [comment]
No, sadly, using that flag didn't work.

However, it seemed to be some kind of incompatibility of my rc.lua and theme.lua with the current AwesomeWM version. I still don't know why this happened, but I managed to fix it.

Some flipped gear.shapes functions were not working as they did previously:
```lua
local flipped_arrow = function(cr, width, height)
    cr:translate(width, height)
    cr:rotate(math.pi)
    gears.shape.rectangular_tag(cr, width, height)
end
```
And now they work:
```lua
local flipped_arrow = function(cr, width, height)
    local shape = gears.shape.transform(gears.shape.rectangular_tag)
        :translate(width, height)
        :rotate(math.pi)
    shape(cr, width, height)
end
```

![Screenshot_2021-10-17-05_1366x768](https://user-images.githubusercontent.com/59768785/137641720-c4e31851-698d-410c-99b3-56590071f49e.png)

(Weird icon sizes on right "taskbar" are still being a problem, but I'll see how to fix them.)

**Comment by Elv13** [comment]
Do you have a minimal example `rc.lua` which can reproduce the problem. I am still interested in getting to the bottom of this. We try to make the releases backward compatible. Obviously, the more elaborate the config is, the more likely it rely in undefined/undocumented behaviors and/or private APIs. But that doesn't look like to be the case if the change you posted ended up fixing the problem.

**Comment by JezerM** [comment]
Well, this would be a minimal config of mine: [see this gist](https://gist.github.com/JezerM/3ff4fc47ff1463488b815712d3f9ca58)

And here are the problematic functions: [see here](https://gist.github.com/JezerM/3ff4fc47ff1463488b815712d3f9ca58#file-powerline-lua-L5-L29)
